client/system_key.go

```diff
@@ -0,0 +1,127 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+)
+
+type SystemKeyMismatchAction string
+
+const (
+	MismatchActionProceed       SystemKeyMismatchAction = "proceed"
+	MismatchActionWaitForChange SystemKeyMismatchAction = "wait-for-change"
+)
+
+var (
+	// ErrMismatchAdviceUnsupported indicates that current snapd does not
+	// support the API to obtain advice on system key mismatch.
+	ErrMismatchAdviceUnsupported = errors.New("system-key mismatch action is not supported")
+)
+
+type AdvisedAction struct {
+	SuggestedAction SystemKeyMismatchAction
+	// ChangeID carries the change to wait for if action was "wait-for-change".
+	ChangeID string
+}
+
+// SystemKeyMismatchAdvise is usually called after having detected a mismatch of
+// the system keys and performs an access to snapd API with the goal of getting
+// an advice from snapd on how to proceed. The request carries system-key
+// derived by the client. Returns an error of type *Error for errors
+// communicated directly through snapd API. Note, that the response is only
+// advisory.
+func (client *Client) SystemKeyMismatchAdvise(systemKey string) (*AdvisedAction, error) {
+	if systemKey == "" {
+		return nil, fmt.Errorf("no system key provided")
+	}
+
+	// same as for /v2/system-info
+	opts := &doOptions{
+		Timeout: 25 * time.Second,
+		Retry:   doRetry,
+	}
+
+	// update maintenance status
+	client.checkMaintenanceJSON()
+
+	var rsp response
+	var body io.Reader
+
+	q := url.Values{}
+	q.Add("system-key-mismatch", "true")
+
+	d, doErr := json.Marshal(struct {
+		SystemKey string `json:"system-key"`
```

Comment:
btw. we could drop the URL parameter (which looks odd), and use this field only, maybe renamed to `system-key-mismatch`